vignettes/tcplfit2-vignette.Rmd

```diff
@@ -362,3 +362,216 @@ The output table resulting from the previous code chunk is the same format as th
+                   res_up2[1, c('Name', "Max Conc.", "bmd", "bmdl", "bmdu")])
+DT::datatable(output_up,rownames = FALSE)
+```
+
+```{r example upper plot}
+# generate some concentration for the fitting curve 
+logc_plot <- seq(from=-3,to=2,by=0.05)
+conc_plot <- 10**logc_plot
+
+# initiate plot
+plot(conc,resp,xlab="conc (uM)",ylab="Response",xlim=c(0.001,500),ylim=c(-5,60),
+       log="x",main=paste(name,"\n",assay),cex.main=0.9)
+# add vertical lines to mark the maximum concentration in the data and the upper threshold set by bmd_up_bnd
+abline(v=max(conc), lty = 2, col = "black", lwd=2)
+abline(v=160, lty = 2, col = "darkviolet", lwd=2)
+
+# add marker for BMD and its boundaries before `censoring`
+lines(c(res_up$bmd,res_up$bmd),c(0,50),col="green",lwd=2)
+rect(xleft=res_up$bmdl,ybottom=0,xright=res_up$bmdu,ytop=50,col=rgb(0,1,0, alpha = .5), border = NA)
+points(res_up$bmd, 0, pch = "x", col = "green")
+
+# add marker for BMD and its boundaries after `censoring`
+lines(c(res_up2$bmd,res_up2$bmd),c(0,50),col="blue",lwd=2)
+rect(xleft=res_up2$bmdl,ybottom=0,xright=res_up2$bmdu,ytop=50,col=rgb(0,0,1, alpha = .5), border = NA)
+points(res_up2$bmd, 0, pch = "x", col = "blue")
+
+# add the fitting curve
+lines(conc_plot, exp4(ps = c(res_low$tp, res_low$ga), conc_plot))
+```
+
+**Figure 9**: This plot shows how BMD and its boundaries changed after "censoring". The green line and green "X" mark the BMD before "censoring" and the green column shows the original boundaries. The blue line and blue "X" mark the "censored" BMD and the blue column shows the updated boundaries. The brown dash line represents the maximum concentration in the data, and the violet dash line represents the threshold value set by `bmd_up_bnd`. In this example, BMD was "censored" to the threshold value, and its boundaries were shifted left.
+
+`bmd_low_bnd` and `bmd_low_bnd` are also available in `tcplhit2_core`. If one chooses to use `tcplfit2_core` and `tcplhit2_core` separately, they can do the same "censoring" for BMD. The results from these two methods are identical. Some example codes are provided below to show how to get the same result from the lower bound example above with `tcplhit2_core`.
+```{r example with hit core}
+# using the same data, fit curves 
+param <- tcplfit2_core(conc2, resp2, cutoff = cutoff)
+hit_res <- tcplhit2_core(param, conc2, resp2, cutoff = cutoff, onesd = onesd, 
+                         bmd_low_bnd = 0.8)
+```
+
+```{r res-hit core}
+# adding the result from tcplhit2_core to the output table for comparison
+hit_res["Name"]<-  paste("Chlorothalonil", "tcplhit2_core", sep = "-")
+hit_res['Min. Conc.'] <- min(conc2)
+hit_res[1, c("Min. Conc.", "bmd", "bmdl", "bmdu")] <- round(hit_res[1, c("Min. Conc.", "bmd", "bmdl", "bmdu")], 3)
+
+output_low <- rbind(output_low, 
+                    hit_res[1, c('Name', "Min. Conc.", "bmd", "bmdl", "bmdu")])
+DT::datatable(output_low,rownames = FALSE)
+```
+
+
```

Comment:
One other 'sub-example' we want to offer is what happens if the estimated BMD is between the "threshold value" and the "reference dose".  That is, what happens to the BMD and confidence bounds if the estimated BMD is less than the lowest experimental dose group, but not lower than the lower bound threshold (lowest dose group * boundary multiplier).

Only need to show the example for one direction.